common/buffers/src/test/java/io/helidon/common/buffers/LazyStringTest.java

```diff
@@ -0,0 +1,74 @@
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class LazyStringTest {
+    @ParameterizedTest
+    @MethodSource("owsData")
+    void testOwsHandling(OwsTestData data) {
+        assertThat(data.string().stripOws(), is(data.expected()));
+    }
+
+    private static Stream<OwsTestData> owsData() {
+        return Stream.of(
+                new OwsTestData(new LazyString("some-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString(" some-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("\tsome-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("  some-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("\t\tsome-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString(" \tsome-value".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("some-value ".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("some-value\t".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("some-value  ".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("some-value\t\t".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("some-value \t".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString(" some-value ".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("\tsome-value\t".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("  some-value  ".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString("\t\tsome-value\t\t".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString(" \tsome-value\t ".getBytes(US_ASCII), US_ASCII), "some-value"),
+                new OwsTestData(new LazyString(" \t\t ".getBytes(US_ASCII), US_ASCII), ""),
+                new OwsTestData(new LazyString(" \t\r\t ".getBytes(US_ASCII), US_ASCII), "\r")
+        );
+    }
+
+    record OwsTestData(LazyString string, String expected) {
+        @Override
+        public String toString() {
+            StringBuilder result = new StringBuilder();
+
+            for (char c : string().toString().toCharArray()) {
+                switch (c) {
+                case '\t' -> result.append("\\t");
+                case '\r' -> result.append("\\r");
+                default -> result.append(c);
+                }
+            }
```

Comment:
Formatting seems little off
```suggestion
            for (char c : string().toString().toCharArray()) {
                switch (c) {
                    case '\t' -> result.append("\\t");
                    case '\r' -> result.append("\\r");
                    default -> result.append(c);
                }
            }
```

Reply:
This is the formatting I get with the Helidon format for Idea